### PR TITLE
Ard chapter05

### PR DESCRIPTION
**Pull request by ardavey**

### CODE REVIEW

File: chapter05/ard/ex3.pl [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+}
+
+# We need a list of source hosts, sorted by hostname
+foreach my $source ( sort keys %data ) {
+  say "$source";
+
+  # And now a list of destination hosts sorted by hostname 
+  foreach my $dest ( sort keys %{ $data{$source} } ) {
+    # The tabs are just for Alex - he loves them!
+    printf( "\t%s\t%u\n", $dest, $data{$source}{$dest} );
```
Comment (Spinodal) [inline chapter05/ard/ex3.pl:23]: now look whos not following the correct format!  looking at the book, i think the destination is only meant to be separated from the bytes by a space

The beginning of the line is a bit ambiguous so i'll let you off :stuck_out_tongue:

Comment (ardavey) [inline chapter05/ard/ex3.pl:23]: Well, the sample data file extract printed in the book was space-separated where the actual data file used tabs - this is me exacting payback on the book :wink:

File: chapter05/ard/ex1.pl [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  occupation => 'Movie Star',
+  real_age => 35,  # Ginger is a liar.
+  hat => undef,
+};
+
+# Repeat offence!
+my %passenger_2 = {
+  name => 'Mary-Ann',
+  age => 19,
+  hat => 'bonnet',  # Nobody over the age of 3 should be allowed to wear a bonnet.
```
Comment (utopia-2) [inline chapter05/ard/ex1.pl:14]: Isn't there a get out clause to that rule, specifically for people named _Mary-Ann_? It just _sounds_ like a bonnet name.

File: chapter05/ard/ex2.pl [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+use strict;
+use warnings;
+
+use 5.010;
+
+my %data = ();
+
+# Bit of a hack - I'm using this as a fake "destination" host to count total traffic
+# from each source. Contains asterisks and spaces so it shouldn't ever clash with a real host name.
+my $any_dest = '* ANYWHERE *';
```
Comment (utopia-2) [inline chapter05/ard/ex2.pl:12]: Might be cleaner to just handle the totals in a separate hash.

Comment (ardavey) [inline chapter05/ard/ex2.pl:12]: Agreed.  I was tempted to make an uber-hash with sub-arrayrefs and hashrefs to contain everything but thought better of it.
Amusingly, the solutions did something similar to this.

File: chapter05/ard/ex1.pl (hunk 2 above)
Comment (ardavey) [inline chapter05/ard/ex1.pl:14]: I may have been following an older version of the ISO hat spec.

old_woman_in_sunbonnet_by_doris_ulmann

**Pull request merged**